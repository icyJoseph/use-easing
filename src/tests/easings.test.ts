import {
  linear,
  easeInQuad,
  easeOutQuad,
  easeInOutSine,
  easeOutElastic,
  easeBackIn,
  easeInOutBack
} from "../easings";

const easings = [
  linear,
  easeInQuad,
  easeOutQuad,
  easeInOutSine,
  easeOutElastic,
  easeBackIn,
  easeInOutBack
];

describe("all easings", () => {
  // t: current time in ms
  // b: start
  // c: change
  // d: duration in ms
  const [b, c, d] = [0, 1, 10];
  const timePoints = Array.from({ length: 10 }, (_, t) => [t, b, c, d]);
  const results = timePoints.map(([w, x, y, z]) =>
    easings.map(easing => easing(w, x, y, z))
  );
  it("start and end at the same point after the same time", () => {
    const [startPoints] = results;
    const [endPoints] = results.slice(-1);
    expect(startPoints.every(point => point === b));
    expect(endPoints.every(point => point === c));
  });

  it("matches known behavior", () => {
    expect(results).toMatchSnapshot();
  });
});
