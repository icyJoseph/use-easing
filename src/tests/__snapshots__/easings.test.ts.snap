// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`all easings matches known behavior 1`] = `
Array [
  Array [
    0,
    0,
    0,
    0,
    0,
    0,
    0,
  ],
  Array [
    0.1,
    0.010000000000000002,
    0.19,
    0.024471741852423234,
    1.25,
    -0.014314220000000002,
    -0.037518552000000004,
  ],
  Array [
    0.2,
    0.04000000000000001,
    0.36000000000000004,
    0.09549150281252627,
    1.125,
    -0.04645056000000001,
    -0.092555656,
  ],
  Array [
    0.3,
    0.09,
    0.51,
    0.20610737385376343,
    0.875,
    -0.08019954000000001,
    -0.07883348399999998,
  ],
  Array [
    0.4,
    0.16000000000000003,
    0.6400000000000001,
    0.3454915028125263,
    1.03125,
    -0.09935168000000003,
    0.0899257920000001,
  ],
  Array [
    0.5,
    0.25,
    0.75,
    0.49999999999999994,
    1.015625,
    -0.08769750000000004,
    0.5,
  ],
  Array [
    0.6,
    0.36,
    0.84,
    0.6545084971874737,
    0.984375,
    -0.02902752000000009,
    0.9100742079999999,
  ],
  Array [
    0.7,
    0.48999999999999994,
    0.9099999999999999,
    0.7938926261462365,
    1.00390625,
    0.09286773999999981,
    1.078833484,
  ],
  Array [
    0.8,
    0.6400000000000001,
    0.96,
    0.9045084971874737,
    1.001953125,
    0.29419776000000003,
    1.092555656,
  ],
  Array [
    0.9,
    0.81,
    0.9900000000000001,
    0.9755282581475768,
    0.998046875,
    0.5911720199999999,
    1.0375185519999999,
  ],
]
`;
